src/shared/Core/HttpClientFactory.cs

```diff
@@ -301,5 +335,68 @@ public bool TryCreateProxy(out IWebProxy proxy)
             proxy = null;
             return false;
         }
+
+        public class CurlCookieParser
+        {
+            private readonly ITrace _trace;
+
+            public CurlCookieParser(ITrace trace)
```

Comment:
We should add a quick check if `content` is null or whitespace here and just return an empty collection.

```csharp
public IList<Cookie> Parse(string content)
{
    if (string.IsNullOrWhiteSpace(content))
    {
        return Array.Empty<Cookie>();
    }

	// rest of method
}
```